### PR TITLE
vmware-horizon-client: remove bundled libpng

### LINKED ISSUE #251148: Vmware-horizon-client crashes when triggering browser/firefox
### Describe the bug
When opening a vmware-view server connection it crashes with message like the below. It appears to be trying to trigger a call to my default browser (firefox) but is unable to start it

XPCOMGlueLoad error for file /nix/store/lxvr36j5wh5x6j3pm0hr6fwg8k7fhpn5-firefox-116.0.3/lib/firefox/libxul.so:
/nix/store/lxvr36j5wh5x6j3pm0hr6fwg8k7fhpn5-firefox-116.0.3/lib/firefox/libxul.so: undefined symbol: png_get_next_frame_delay_num, version PNG16_0
Couldn't load XPCOM.

### Steps To Reproduce
1. start vmware-view
2. Try to launch a vmware-view server connection, for example desktopauth.citi.com
3. Crash

### Expected behavior
Not crashing :)

### Notify maintainers
@flokli @buckley310 

### Metadata

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.4.11, NixOS, 23.05 (Stoat), 23.05.20230822.a16f7eb`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.5`
 - nixpkgs: `/nix/store/jzj4v4cm2z5l40bi7pi36fv1vf6i68d3-source`
```

### PR DESCRIPTION
**Pull request by buckley310**
## Description of changes
fixes #251148

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by buckley310** [comment]
@das-g I think this might resolve the web-based sign-on issues. It at least allows the browser to launch and open a sign-in screen. Feel like giving it a try?

**Comment by das-g** [comment]
Result of `nixpkgs-review pr 313250` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vmware-horizon-client</li>
  </ul>
</details>

**Comment by das-g** [comment]
This indeed fixes #251148 for me. :+1:

File: pkgs/applications/networking/remote/vmware-horizon-client/default.nix [hunk 1]
```diff
@@ -54,6 +54,9 @@ let
       # Deleting the bundled library is the simplest way to force it to use our version.
       rm "$out/lib/vmware/gcc/libstdc++.so.6"
 
+      # This bundled version of libpng causes browser issues, and would prevent web-based signon
```
Comment (das-g) [inline pkgs/applications/networking/remote/vmware-horizon-client/default.nix:57]:
```suggestion
      # This bundled version of libpng causes browser issues, and would prevent web-based sign-on.
```

**Review by das-g — CHANGES_REQUESTED** [review]

**Comment by buckley310** [comment]
Took suggested change and squashed.

**Comment by das-g** [comment]
You should probably put
```
fixes #251148
```
into the commit message, the PR description or both, so that merging this will close the bug report. (Or do you expect this change to fix the reported problem only for some cases, but not all?)

**Comment by buckley310** [comment]
Thanks. I forgot to add the description when I marked it as not a draft.

**Review by das-g — APPROVED** [review]
👍 change LGTM
👍 commit LGTM
✔️ `vmware-view` seems to work no worse¹ than without this change
:white_check_mark: fixes #251148
:shrug: (I didn't test `vmware-usbarbitrator`)

:shipit: Ship it!

---
¹some issues I observed _regardless_ of this change (i.e., they happen the same with and without this change):
- When connecting from within my employer's network to my employer's VDI, a native login dialog rather than a browser window is used. Text typed into the username and password fields isn't always visible and if so, the login button looks like it continues to be disabled, but works anyway.
- After shutting down a remote session, closing the window listing possible servers to connect to leads to the program exiting with the following terminal output:
   ```
   viewusb.xml:1: parser error : Extra content at the end of the document
   " name="max-concurrent-connections">0</stat></add-client-info></client-info-usb>
                                                                                  ^
   /nix/store/vh3p18frm16q6y5zqcrf28mpwq2xcbd1-vmware-horizon-files-2312.1/bin/vmware-view: line 86: 70205 Segmentation fault      (core dumped) "$binPath/bin/$binFile" "$@"
   
   ```
   and exit code `139`.

**Comment by das-g** [comment]
> some issues I observed _regardless_ of this change [...]

(I'll report those separately when I get to it.)

**Pull request merged**